src/pages/suggestions/new/index.tsx

```diff
@@ -1,92 +1,101 @@
+              }
 
-        setInstrumentList(response.data as Instrument[])
-      })
-      .catch(() => {
-        setShowLoadingError(true)
+              router.push("/suggestions").then(() => {
+                onSuccess()
+                dispatch(updateNewSuggestion(initialState.suggestion))
+                dispatch(setActiveArea(Area.SongInformation))
+              })
+            })
+            .catch(() => onError())
+        })
+        .catch(() => onError())
+    } else {
+      onError()
+    }
+  }
+
+  const onAreaSelect = (area: Area) => {
+    dispatch(setActiveArea(area))
+  }
+
+  const onInstrumentSubmit = (newInstruments: NewSuggestionInstrument[]) => {
+    dispatch(
+      updateNewSuggestion({
+        ...suggestion,
+        instruments: newInstruments,
       })
-      .finally(() => {
-        setShowSpinner(false)
+    )
+    dispatch(setActiveArea(Area.Review))
+  }
+
+  const onSongInformationSubmit = ({ title, artist, link, motivation }: InputsSongInformation) => {
+    dispatch(
+      updateNewSuggestion({
+        ...suggestion,
+        title,
+        artist: [artist],
+        link,
+        motivation,
       })
-  }, [supabaseClient])
+    )
+  }
 
   return (
-    <FormProvider {...methods}>
-      <div className={"page-wrapper"}>
-        <div className={"flex justify-between"}>
-          <div className={"page-header"}>New Suggestion</div>
-          <XMarkIcon
-            data-cy={"button-discard-new-suggestion"}
-            className={"h-8 w-8 cursor-pointer text-zinc-400 hover:text-red-500"}
-            onClick={() => router.push("/suggestions")}
-          />
-        </div>
-
-        {showSpinner && (
-          <div className={"h-[75vh] text-center"} data-cy="suggestions-spinner">
-            <Spinner size={10} />
-          </div>
-        )}
-        {showLoadingError && (
-          <div className={"mt-6"} data-cy="failed-fetching-instruments">
-            <ErrorPopup
-              text={`“Something went wrong”
-            You can try again. Contact support if this error persists.`}
-              closePopup={() => {}}
-            />
-          </div>
-        )}
-        {!showLoadingError && (
-          <div className={"mx-auto text-center lg:w-2/4"}>
-            <ProgressBar />
-            {activeArea == Area.SongInformation && <SongInformationArea />}
-            {activeArea == Area.Instruments && <InstrumentsArea instrumentList={instrumentList} />}
-            {activeArea == Area.Review && <ReviewArea />}
-          </div>
-        )}
-      </div>
-    </FormProvider>
+    <SuggestionPageSection
+      title={"New Suggestion"}
+      newSuggestion={suggestion}
+      startingArea={activeArea}
+      onSongInformationSubmit={onSongInformationSubmit}
+      onAreaSelect={onAreaSelect}
+      onInstrumentSubmit={(newInstruments) => {
+        onInstrumentSubmit(newInstruments)
+      }}
+      onReviewSubmit={(success, error) => saveSuggestion(success, error)}
```

Comment:
I'm so confused on what the success & error parameters do, could you explain this to me?